src/processTargets/PipelineStages.types.ts

```diff
@@ -0,0 +1,10 @@
+import { Target } from "../typings/target.types";
+import { ProcessedTargetsContext } from "../typings/Types";
+
+export interface MarkStage {
+  run(context: ProcessedTargetsContext): Target[];
+}
+
+export interface ModifierStage {
+  run(context: ProcessedTargetsContext, target?: Target): Target[];
```

Comment:
Why the `?` on the `target` arg?  When would this one be undefined?

Reply:
true